### ISSUE TITLE
Web staging builds are failing with latest commits and build an outdated version

### ISSUE THREAD

**Issue by NathanLovato** [GDQuest/learn-gdscript#328]
See https://github.com/GDQuest/learn-gdscript/runs/5215629805?check_suite_focus=true for the latest attempt. Error message:

```sh
Run ./run push web
`build/web` does not seem to exist. Did you type the name correctly?
Error: Process completed with exit code 1.
```

Right now, an outdated version of the app is live on staging.

**Comment by NathanLovato** [comment]
Fixed by dee9e44acacc720dea042a9d3899936d0fedbc3b

**Comment by Xananax** [comment]
Ok, it's fixed. 

Some documentation for what I've made, that I'm not sure where to write.

## Change 1: `run` helper

There's a small bash script at the project root, which helps building, testing, and cleaning up. I needed this to simplify the CI setup (so I could test more things locally), but also to be able to edit quickly the web build.

A few examples:

- `./run export:windows` will build the windows export
- `./run web:watch` will watch the directory for changes, and will recompile the web app on save, and copy the static files when you save those
- `./run web:debug` will export a debug web build, start web-server, and watch files for changes
- `./run clean` will remove the build directory
- `./run help` will list all commands, and more info can be obtained from each, for example: `./run web:debug help`


## Change 2: CI 

The CI will autostart from `release` and `staging`. The behavior is as follows:

- last commit gets exported for Linux, Windows. MacOSX, and web
- each gets sent to itch, tagged with the name of the branch (`release`, or `staging`)
- each gets saved to Github as artifacts (which are not, at the moment, accessible to people)
- each web build gets pushed to `github pages`, in a subdirectory named like the branch

The important part to understand is that while the CI auto-triggers from `release` and `staging`, it can be manually triggered _from any branch_. That means if we make a special build that we call `special`, and run the CI on that, we'll get files tagged `special` on itch, and a subdirectory `gdquest.github.io/learn-gdscript/special`. This should allow for easily distributing experiments
